### PR TITLE
feature: more Shl+Shr impls

### LINKED ISSUE #327: Move `Shl` / `Shr` impls out of support into main lib
It would be nice to support the [Shl](https://doc.rust-lang.org/std/ops/trait.Shl.html) and Shr traits for uints.

### PR DESCRIPTION
**Pull request by prestwich**
## Motivation

Closes #327 

## Solution

Add Shl and Shr implementations for types that are losslessly castable to usize by re-using the existing `usize` implementation

## PR Checklist

- [ ] Added Tests
- [ ] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

**Review by gakonst — APPROVED** [review]
good w me, simple enough

File: src/bits.rs [hunk 1]
```diff
@@ -548,105 +548,79 @@ impl_bit_op!(BitOr, bitor, BitOrAssign, bitor_assign);
...
 
-impl<const BITS: usize, const LIMBS: usize> Shr<&usize> for &Uint<BITS, LIMBS> {
-    type Output = Uint<BITS, LIMBS>;
-
-    #[inline(always)]
-    fn shr(self, rhs: &usize) -> Self::Output {
-        self.wrapping_shr(*rhs)
-    }
-}
+impl_shift!(usize, u8, u16, u32, i8, i16, i32, isize);
```
Comment (DaniPopes) [inline src/bits.rs:623]: `u64`/`i64`?

**Review by DaniPopes — APPROVED** [review]

**Review by Evalir — APPROVED** [review]
good w/ me! pending dani's nit

File: src/bits.rs [hunk 2]
```diff
@@ -620,7 +698,11 @@ macro_rules! impl_shift {
     };
 }
 
-impl_shift!(usize, u8, u16, u32, u64, isize, i8, i16, i32, i64);
+impl_shift!(usize, u8, u16, u32, isize, i8, i16, i32);
+
+// Only when losslessy castable to usize.
+#[cfg(target_pointer_width = "64")]
+impl_shift!(u64, i64);
```
Comment (DaniPopes) [inline src/bits.rs:705]: below is lossless too, it just pads

**Pull request merged**